File: minegame-server/src/main/java/top/nipuru/minegame/server/processor/PlayerPrivateChatServerProcessor.java
package top.nipuru.minegame.server.processor;

import com.alipay.remoting.BizContext;
import com.alipay.remoting.rpc.protocol.SyncUserProcessor;
import top.nipuru.minegame.common.message.PlayerChatMessage;
import top.nipuru.minegame.common.message.PlayerPrivateChatMessage;
import top.nipuru.minegame.server.BukkitPlugin;
import top.nipuru.minegame.server.game.player.GamePlayer;
import org.bukkit.Bukkit;

public class PlayerPrivateChatServerProcessor extends SyncUserProcessor<PlayerPrivateChatMessage> {

    private final BukkitPlugin plugin;

    public PlayerPrivateChatServerProcessor(BukkitPlugin plugin) {
        this.plugin = plugin;
    }

    @Override
    public Object handleRequest(BizContext bizContext, PlayerPrivateChatMessage request) {
        try {
            GamePlayer player = plugin.getPlayer(Bukkit.getPlayerExact(request.getReceiver()));
            if (!player.getChatManager().couldReceivePrivate(request.getSender())) {
                return PlayerPrivateChatMessage.DENY;
            }
            player.getChatManager().receivePrivate(request.getSender(), request.getFragments());
        } catch (Exception ignored) {} // 有概率玩家在跨服或者离线
        return PlayerPrivateChatMessage.SUCCESS;
    }


    @Override
    public String interest() {
        return PlayerChatMessage.class.getName();
    }

}

File: minegame-game/src/main/kotlin/top/nipuru/minegame/game/processor/PlayerPrivateChatServerProcessor.kt
package top.nipuru.minegame.game.processor

import com.alipay.remoting.BizContext
import com.alipay.remoting.rpc.protocol.SyncUserProcessor
import org.bukkit.Bukkit
import top.nipuru.minegame.common.message.PlayerChatMessage
import top.nipuru.minegame.common.message.PlayerPrivateChatMessage
import top.nipuru.minegame.game.gameplay.player.GamePlayer
import top.nipuru.minegame.game.playerList

class PlayerPrivateChatServerProcessor : SyncUserProcessor<PlayerPrivateChatMessage>() {
    override fun handleRequest(bizContext: BizContext, request: PlayerPrivateChatMessage): Any {
        try {
            val bukkitPlayer = Bukkit.getPlayerExact(request.receiver)
            val player: GamePlayer = playerList.getPlayer(bukkitPlayer!!.uniqueId)
            val manager = player.chatManager
            if (!manager.couldReceivePrivate(request.sender)) {
                return PlayerPrivateChatMessage.DENY
            }
            manager.receivePrivate(request.sender, request.fragments)
        } catch (ignored: Exception) {
        } // 有概率玩家在跨服或者离线

        return PlayerPrivateChatMessage.SUCCESS
    }


    override fun interest(): String {
        return PlayerChatMessage::class.java.name
    }
}
